src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -308,14 +308,14 @@ class SamRecordClipper(val mode: ClippingMode, val autoClipAttributes: Boolean)
       // Pick the mid point in the reference window over which the record and its mate overlap.  The read ends at the
       // mid point, or if the mid point is in a deletion, the base prior to the deletion.  The mate ends at the mid
       // point, or if the mid point is in a deletion, the base after the deletion.
-      val midPoint  = (rec.start + mate.end) / 2
+      val midPoint = (mate.start + rec.end) / 2
```

Comment:
the unspoken rule in this repo is to align the equals, sorry for the nit!
```suggestion
      val midPoint  = (mate.start + rec.end) / 2
```

Reply:
Also, I like your alternative as _I think_ it does a better job of balancing the # of bases trimmed off of each end of the read, which is where the low quality bases are more likely anyhow.

```scala
val midPoint  = { // Need to ensure the midpoint actually occurs in the overlapped region
  val retval = (rec.start + mate.end) / 2
  if (retval > rec.end) rec.end
  else if (retval < mate.start) mate.start
  else retval
}
```